### DISCUSSION TITLE
Firefly Data Importer: Could not store transaction

### DISCUSSION THREAD

**Discussion by asmartin** [firefly-iii/firefly-iii#8139] [Firefly III] [answered]
Hello,

I installed Firefly 6.0.23 as well as Firefly Data Importer 1.3.8. I'm attempting to import a CSV file; the import appears to succeed (I see the transactions in Firefly), but the importer fails with the following error:
```
[2023-11-07 22:01:54] local.DEBUG: Going to add import tag to transaction group #71  
[2023-11-07 22:01:54] local.ERROR: Connection refused for URI http://mydomain.com/api/v1/transactions/71  
[2023-11-07 22:01:54] local.DEBUG: Add error on index #0 (line no. 1): Could not store transaction: see the log files.
```

I set `APP_LOG_LEVEL=debug` but still can't get it to print any messages about this error on the Firefly side; it appears to be working successfully. What can I do to debug this error in the Firefly Data Importer (or should I just ignore it)?

Thanks!

**Comment by JC5** [comment]
Well the error says "Connection refused for URI http://mydomain.com/api/v1/transactions/71", that seems pretty straight-forward to me.

**Comment by asmartin** [comment]
I agree, yet I can see the transactions showing up in Firefly. If I curl that URL directly, I get a "not authorized" error, which makes sense, but clearly the OAuth token I'm using with the Firefly Data Importer is functional since the transactions are added. What can I do to debug this further?

**Reply by JC5** [reply, marked as answer]
This is the second connection to Firefly III, when the importer tries to connect the import tag to the imported transaction. I have no idea why it fails, but you could try disabling the import tag in the configuration of the import.

**Reply by asmartin** [reply]
Thanks, that worked! Digging into it more, I determined the reason it was failing - my nginx config did not allow the `PUT` request type. Once I added it as an allowed request type, it worked even when import tags were enabled. Thank you again for the help!
